presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -256,21 +254,33 @@ public void recordFinish()
         return revocableMemoryFuture.get();
     }
 
-    public void reserveMemory(long bytes)
+    // caller should close this context as it's a new context
+    public LocalMemoryContext newLocalSystemMemoryContext()
```

Comment:
> caller should close this context as it's a new context

Currently, this is not always the case. See for example `ScanFilterAndProjectOperator`